src/node.rs

```diff
@@ -233,13 +240,29 @@ impl Node {
             onion_message_handler: Arc::new(IgnoringMessageHandler {}),
         };
 
+        let lsp_client_pubkey = lsp_client.clone().map(|lsp| lsp.pubkey);
+
+        // save connection info peer connection list to auto connect
+        if let Some(lsp) = lsp_client.clone() {
+            match persister.persist_peer_connection_info(
+                lsp_client_pubkey.unwrap().to_string(),
+                lsp.connection_string.clone(),
+            ) {
```

Comment:
can probably make `persist_peer_connection_info` take `&str`s so we don't need to do all these clones